OPHD/UI/IconGrid.cpp

```diff
@@ -371,11 +371,61 @@ void IconGrid::update()
 		// Name Tooltip
 		if (mShowTooltip)
 		{
-			const auto& highlightedName = mIconItemList[mHighlightIndex].name;
-			const auto tooltipRect = NAS2D::Rectangle<int>{{position.x, position.y - 15}, {mFont.width(highlightedName) + 4, mFont.height()}};
-			renderer.drawBoxFilled(tooltipRect, NAS2D::Color{245, 245, 245});
-			renderer.drawBox(tooltipRect, NAS2D::Color{175, 175, 175});
-			renderer.drawText(mFont, highlightedName, position + NAS2D::Vector{2, -15}, NAS2D::Color::Black);
+			std::vector<std::string> toolTipStrings;
+			toolTipStrings.push_back(mIconItemList[mHighlightIndex].name);
+			/* HEADS UP Feels like there is a better way. Unless more robots are to be added it works.
+			* Without this name check accessing the structure resource and population requirements causes an error because the robots are not structures*/
+			if (toolTipStrings[0] != "Robodigger" && toolTipStrings[0] != "Robodozer" && toolTipStrings[0] != "Robominer" && this->mIconSheet.name() != "ui/factory.png")
```

Comment:
This seems a bit too ad hoc. I would avoid using display strings, which are subject to change, for any kind of logical condition.

There's probably a better way to handle this case, though I may need to circle back to this. Perhaps we need a new field on each data element for use in the tooltip message. It would probably be easier to get this data when setting the grid icon then when trying to display it.